static/app/utils/list/useListItemCheckboxState.tsx

```diff
@@ -22,97 +31,166 @@ interface Props {
   queryKey: undefined | ApiQueryKey | InfiniteApiQueryKey;
 }
 
+interface InternalProps {
+  setState: Dispatch<SetStateAction<State>>;
+  state: State;
+}
+type MergedProps = PublicProps & InternalProps;
+
 /**
  * We can either have a list of ids, or have all selected.
  * When all is selected we may, or may not, have all ids loaded into the browser
  */
 type State = {ids: Set<string>} | {all: true};
 
-interface Return {
+interface Return extends PublicProps {
   /**
    * How many ids are selected
    */
   countSelected: number;
+
   /**
    * Ensure nothing is selected, no matter the state prior
    */
   deselectAll: () => void;
+
   /**
    * True if all are selected
    *
    * When some are selected returns 'indeterminate'
+   *
+   * Useful at the top of a list of items: <Checkbox checked={isAllSelected}/>
    */
   isAllSelected: 'indeterminate' | boolean;
+
   /**
    * True if one or more are selected
+   *
+   * Useful to show bulk-actions at the top of a list whenever anything is selected
    */
   isAnySelected: boolean;
+
   /**
    * True if this specific id is selected
+   *
+   * Useful for individual list items: <Checkbox checked={isSelected(id) !== false}/>
    */
   isSelected: (id: string) => 'all-selected' | boolean;
+
   /**
    * Record that all are selected, wether or not all feedback ids are loaded or not
    */
   selectAll: () => void;
+
   /**
    * The list of specifically selected ids, or 'all' to save space
    */
   selectedIds: 'all' | string[];
+
   /**
    * Toggle if a feedback is selected or not
    * It's not possible to toggle when all are selected, but not all are loaded
    */
-  toggleSelected: (id: string) => void;
+  toggleSelected: (id: string | string[]) => void;
 }
 
-export default function useListItemCheckboxState({
+const ListItemCheckboxContext = createContext<{
+  hits: number;
+  knownIds: string[];
+  queryKey: undefined | ApiQueryKey | InfiniteApiQueryKey;
+  setState?: Dispatch<SetStateAction<State>>;
+  state?: State;
+}>({
+  hits: 0,
+  knownIds: [],
+  queryKey: undefined,
+});
+
+/**
+ * @public
+ */
```

Comment:
public for knip. gonna remove this comment as part of https://github.com/getsentry/sentry/pull/95189